### PR TITLE
fix shellcheck in k8s.io/apiextensions-apiserver

### PR DESCRIPTION
**Pull request by danielqsj**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

fix shellcheck failures in `staging/src/k8s.io/apiextensions-apiserver`

**Which issue(s) this PR fixes**:

work towards #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by BenTheElder — APPROVED** [review]
/lgtm

**Comment by danielqsj** [comment]
@BenTheElder fixed quota style. PTAL

**Comment by danielqsj** [comment]
@sttts @fejta if you have time, can you help review this? Thanks

File: staging/src/k8s.io/apiextensions-apiserver/hack/build-image.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ function cleanup {
 }
 trap cleanup EXIT
 
-pushd "${KUBE_ROOT}/vendor/k8s.io/apiextensions-apiserver"
+pushd "${KUBE_ROOT}/vendor/k8s.io/apiextensions-apiserver" || exit
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/hack/build-image.sh]: this should be a `exit 1`

**Comment by sttts** [comment]
We have many more of these `*-codegen.sh` files. Can you fix all at once?

File: staging/src/k8s.io/apiextensions-apiserver/hack/build-image.sh (hunk 1 above)
Comment (fejta) [reply staging/src/k8s.io/apiextensions-apiserver/hack/build-image.sh]: put `set -o errexit` at the top of this file (and the other options, like most of our scripts do

File: staging/src/k8s.io/apiextensions-apiserver/hack/verify-codegen.sh [hunk 2]
```diff
@@ -18,8 +18,7 @@ set -o errexit
 set -o nounset
 set -o pipefail
```
Comment (fejta) [inline staging/src/k8s.io/apiextensions-apiserver/hack/verify-codegen.sh]: these

**Comment by danielqsj** [comment]
@sttts @fejta fixed, PTAL

File: staging/src/k8s.io/apiextensions-apiserver/hack/build-image.sh (hunk 1 above)
Comment (fejta) [reply staging/src/k8s.io/apiextensions-apiserver/hack/build-image.sh]: Remove the `|| exit`, otherwise it won't fail

Comment (danielqsj) [reply staging/src/k8s.io/apiextensions-apiserver/hack/build-image.sh]: thanks, removed it and also passed shellcheck. Cool :)

**Pull request merged**